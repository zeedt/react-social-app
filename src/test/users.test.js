import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
import React from "react";
import Users from '../user/user';


configure({ adapter: new Adapter() });
const mockedState = {
  onlineUsers: [
    { username: 'zeed', name: 'Olamide olubayo' },
    { username: 'Bayo', name: 'Bayonle olubayo' }
  ],
  chats: { 'zeed': [] },
  currentUserChattingWith: '',
  open: false
}
describe('User component test', () => {

  it('should be able to load page successfully', () => {
    const wrapper = renderer.create(<Users />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have available users in the card-header class', () => {
    const component = mount(<Users />);
    expect(component.find('.card-header').text()).toEqual('Available Users');
  });

  it('should ensure no user image is rendered when no user is currently online', () => {
    const component = mount(<Users />);
    expect(component.find('.user-chat-image').length).toEqual(0);
  });


  it('should list the users passed as state', () => {
    const component = mount(<Users />);
    component.setState({ ...mockedState });
    expect(component.find('.user-chat-image').length).toEqual(2);
  });

  it('ensure chatbox is not opened when user has not been clicked', () => {
    const component = mount(<Users />);
    component.setState({ ...mockedState });
    expect(component.state.open).not.toBeTruthy();
  });

  it('ensure chatbox is opened when a user has been clicked', () => {
    const component = mount(<Users />);
    component.setState({ ...mockedState });
    component.find('.user-chat-image').at(0).simulate('click')
    expect(component.state().open).toBeTruthy();
  });

  it('ensure the username on the chatbox title is same as the user clicked', () => {
    const component = mount(<Users />);
    component.setState({ ...mockedState });
    component.find('.user-chat-image').at(0).simulate('click')
    expect(component.find('.title').text()).toEqual(component.state().currentUserChattingWith);
  });

});
