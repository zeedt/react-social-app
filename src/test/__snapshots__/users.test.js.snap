// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component test should be able to load page successfully 1`] = `
<div>
  <div
    className="user-list-div card"
  >
    <div
      className="card-header"
    >
      Available Users
    </div>
    <div
      className="user-list-card-body card-body"
    />
  </div>
  <div
    aria-expanded={null}
    className="collapse-view collapse"
    id="example-fade-text"
  >
    <div
      className="chat"
    >
      <div
        className="title"
      >
        
      </div>
      <div
        className="bottomChatDiv text"
        id="bottomChatDiv"
      />
      <div
        className=""
      >
        <div
          className="input-group mb-3"
        >
          <input
            className="form-control "
            name="message"
            onChange={[Function]}
            placeholder="Type message here..."
            type="text"
            value=""
          />
          <div
            className="input-group-append"
            onClick={[Function]}
          >
            <span
              className="input-group-text ficon"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-arrow-right fa-w-14 "
                color="white"
                data-icon="arrow-right"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
